lib/fluent/plugin/parser_syslog.rb

```diff
@@ -103,51 +119,116 @@ def parse(text)
 
       def parse_auto(text, &block)
         if REGEXP_DETECT_RFC5424.match(text)
-          @regexp = @with_priority ? REGEXP_RFC5424_WITH_PRI : REGEXP_RFC5424_NO_PRI
+          @regexp = RFC5424_WITHOUT_TIME_AND_PRI_REGEXP
           @time_parser = @time_parser_rfc5424
           @support_rfc5424_without_subseconds = true
-          parse_plain(text, &block)
+          parse_rfc5424_regex(text, &block)
         else
-          @regexp = @with_priority ? REGEXP_WITH_PRI : REGEXP
+          @regexp = RFC3164_WITHOUT_TIME_AND_PRI_REGEXP
           @time_parser = @time_parser_rfc3164
           if @regexp_parser
-            parse_plain(text, &block)
+            parse_rfc3164_regex(text, &block)
           else
             parse_rfc3164(text, &block)
           end
         end
       end
 
-      def parse_plain(text, &block)
-        m = @regexp.match(text)
-        unless m
+      def parse_rfc3164_regex(text, &block)
+        idx = 0
+        record = {}
+
+        if @with_priority
+          if RFC3164_PRI_REGEXP.match?(text)
+            v = text.index('>')
+            record['pri'] = text[1..v].to_i # trim `<` and ``>
+            idx = v + 1
+          else
+            yield(nil, nil)
+            return
+          end
+        end
+
+        i = idx - 1
+        sq = false
+        @space_count.times do
+          while text[i + 1] == ' '.freeze
+            sq = true
+            i += 1
+          end
+
+          i = text.index(' '.freeze, i + 1)
+        end
+
+        time_str = sq ? text.slice(idx, i - idx).squeeze(' ') : text.slice(idx, i - idx)
+        time = @mutex.synchronize { @time_parser.parse(time_str) }
+        record['time'] = time_str
+
+        parse_plain(time, text, i + 1, record, RFC3164_CAPTURES, &block)
+      end
+
+      def parse_rfc5424_regex(text, &block)
+        idx = 0
+        record = {}
+
+        if @with_priority
+          if (m = RFC5424_PRI_REGEXP.match(text))
+            record['pri'] = m['pri']
+            record['version'] = m['version']
+            idx = m.end(0)
+          else
+            yield(nil, nil)
+            return
+          end
+        end
+
+        i = idx - 1
+        sq = false
+        @space_count_rfc5424.times {
+          while text[i + 1] == ' '.freeze
+            sq = true
+            i += 1
+          end
+
+          i = text.index(' '.freeze, i + 1)
+        }
+
+        time_str = sq ? text.slice(idx, i - idx).squeeze(' '.freeze) : text.slice(idx, i - idx)
+        time = @mutex.synchronize do
+          begin
+            @time_parser.parse(time_str)
+          rescue Fluent::TimeParser::TimeParseError => e
+            if @support_rfc5424_without_subseconds
+              log.trace(e)
+              @time_parser_rfc5424_without_subseconds.parse(time_str)
+            else
+              raise
+            end
+          end
+        end
+
+        record['time'] = time_str
```

Comment:
`record['time'] = time_str if @keep_time_key` is better than unnecessary `insert/delete` 'time' field.

Reply:
fixed https://github.com/fluent/fluentd/pull/2886/commits/5510b8f556b78cb9f268c01a1d08aef3bd7ed4cc